controllers/openstacknet_controller.go

```diff
@@ -192,7 +182,39 @@ func (r *OpenStackNetReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			return ctrl.Result{}, err
 		}
 
+		// get bridgeName from desiredState
 		bridgeName := gjson.Get(string(desiredState["desiredState"]), "interfaces.#.name").Array()[0]
+
+		// Don't add a uniq  label here because the NCP might be related to multiple networks/NetworkAttachmentDefinitions
+		labelSelector := map[string]string{
+			common.OwnerNameSpaceLabelSelector:      instance.Namespace,
+			common.OwnerControllerNameLabelSelector: openstacknet.AppLabel,
+		}
+
+		// generate NodeNetworkConfigurationPolicy for the bridge
+		ncp := common.NetworkConfigurationPolicy{
+			Name:                           bridgeName.String(),
+			Labels:                         labelSelector,
+			NodeNetworkConfigurationPolicy: instance.Spec.AttachConfiguration.NodeNetworkConfigurationPolicy,
+		}
+		err = common.CreateOrUpdateNetworkConfigurationPolicy(r, instance, instance.Kind, &ncp)
```

Comment:
This is where your comment regarding thrashing could occur, right?  If two or more `OsNet`s use the same bridge but attach a different interface in the `desiredState`?

Reply:
yes correct. if different `desiredState` configs get used for networks, but have the same `bridgeName` this would end up in an nncp update loop. An idea could be to check for existing nncp and skip updating the current running nncp config and just log.

Reply:
Could we create an up front validation on OsNets to guard against this case (same bridge, but different interfaces)? Like a webhook?